tests/initcoder_test.go

```diff
@@ -0,0 +1,191 @@
+			receipt, err := createContractWithCodeLenAndGas(net, variant, codeLen, sufficientGas)
+			require.NoError(err)
+			require.Equal(types.ReceiptStatusSuccessful, receipt.Status, "failed to create contract with code length ", codeLen)
+			cost, err := getCreateCost(contract, receipt)
+			require.NoError(err)
+			return cost
+		}
+
+		// since memory is expanded in words of 32 bytes, we want to check that the cost is proportional to the number of words.
+		// hence 30 bytes fit in 1 word, 42 in 2 words and 90 in 3 words.
+		cost1Word := createAndGetCost(30)
+		cost2Words := createAndGetCost(42)
+		cost3Words := createAndGetCost(90)
+
+		require.Equal(wordCost, cost2Words-cost1Word, "cost difference between 1 and 2")
+		require.Equal(wordCost*2, cost3Words-cost1Word, "cost difference between 1 and 3")
+	})
+
+	t.Run("fails without enough gas", func(t *testing.T) {
+		// 4 for a zero byte, 1 to make it fail.
+		receipt, err := createContractWithCodeLenAndGas(net, variant, 1, gasForContract-wordCost-1)
+		require := require.New(t)
+		require.NoError(err)
+		require.Equal(types.ReceiptStatusFailed, receipt.Status,
+			"unexpectedly succeeded to create contract without enough gas")
+	})
+
+	t.Run("with max init code size", func(t *testing.T) {
+		receipt, err := createContractWithCodeLenAndGas(net, variant, MAX_INIT_CODE_SIZE, sufficientGas)
+		require := require.New(t)
+		require.NoError(err)
+		require.Equal(types.ReceiptStatusSuccessful, receipt.Status,
+			"failed to create contract with code length ", MAX_INIT_CODE_SIZE)
+	})
+
+	t.Run("aborts with init code size larger than MAX_INITCODE_SIZE", func(t *testing.T) {
+		receipt, err := createContractWithCodeLenAndGas(net, variant, MAX_INIT_CODE_SIZE+1, sufficientGas)
+		require := require.New(t)
+		require.NoError(err)
+		require.Equal(types.ReceiptStatusFailed, receipt.Status,
+			"unexpectedly succeeded to create contract with init code length greater than MAX_INITCODE_SIZE")
+	})
+}
+
+func testForTransaction(t *testing.T, net *IntegrationTestNet) {
+	t.Run("charges depending on the init code size", func(t *testing.T) {
+		require := require.New(t)
+		// transactions charge 4 gas for each zero byte in data.
+		const zeroByteCost uint64 = 4
+		// create a transaction with 1 byte of code.
+		receipt1, err := runTransactionWithCodeSizeAndGas(t, net, 1, sufficientGas)
+		require.NoError(err)
+		require.Equal(types.ReceiptStatusSuccessful, receipt1.Status,
+			"failed on transfer to empty receiver with valid code")
+
+		// createa a transaction with 2 byte of code.
+		receipt2, err := runTransactionWithCodeSizeAndGas(t, net, 2, sufficientGas)
+		require.NoError(err)
+		require.Equal(types.ReceiptStatusSuccessful, receipt2.Status,
+			"failed on transfer to empty receiver with valid code")
+
+		difference := receipt2.GasUsed - receipt1.GasUsed
+		require.Equal(difference, zeroByteCost,
+			"gas difference between 1 and 2 words should be 4, instead got", difference)
```

Comment:
In the rest of this test you state that code size is measured in bytes, but here it is words. I guess it should be bytes here as well.